### PR TITLE
[RDY] Doc cleanup (4) - clipboard docs + various

### PR DESCRIPTION
**Pull request by ghost**
I have some questions about the clipboard section in `gui_x11.txt`, which I've mostly transferred to `nvim_clipboard.txt`. I think some of it's helpful, but I'm not sure all of it's still applicable.

cc @bfredl

### CODE REVIEW

File: runtime/doc/nvim_clipboard.txt [hunk 1]
```diff
@@ -29,4 +35,77 @@ following option:
 See 'clipboard' for details and more options.
 
 ==============================================================================
+2. X11 selection mechanism				    *nvim-clipboard-x11*
```
Comment (justinmk) [inline runtime/doc/nvim_clipboard.txt]: :+1: to moving this, but the tag should remain `*x11-selection*`. I don't know why we're prefixing a bunch of tags with `nvim-`, I think we need to reconsider that practice...

Comment (Pyrohh) [inline runtime/doc/nvim_clipboard.txt]: > but the tag should remain *x11-selection*

sure

> I don't know why we're prefixing a bunch of tags with nvim-, I think we need to reconsider that practice...

I think it makes sense in certain cases, such as namespacing headers (e.g., `remote-plugin-(intro|hosts|example|manifest)`, but yeah, I agree it's not needed in this case.

**Comment by Pyrohh** [comment]
@justinmk repushed

**Comment by lucc** [comment]
(Sorry I commented on the commits above and not on the diff of this PR so you don't see the lines here.)

File: runtime/doc/nvim_clipboard.txt [hunk 2]
```diff
@@ -29,4 +35,77 @@ following option:
...
+- Select some text in Vim and do "+y.  Go to your browser, select some text in
+  a textfield by dragging with the mouse.  Now use the right mouse button and
+  select "Paste" from the popup menu.  The selected text is overwritten by the
+  text from Vim.
+Note that the text in the "+ register remains available when making a Visual
+selection, which makes other text available in the "* register.  That allows
+overwriting selected text.
+								*x11-cut-buffer*
+There are, by default, 8 cut-buffers: CUT_BUFFER0 to CUT_BUFFER7.
+Vim only uses CUT_BUFFER0, which is the one that xterm uses by default.
```
Comment (bfredl) [inline runtime/doc/nvim_clipboard.txt]: It don't seem that `xsel` uses the cut-buffer as a fallback. Nor I think that we want to. Instead, to keep any owned selection alive after nvim exists, we should detach the clipboard owning `xsel` job ( or alternatively relaunch `xsel` in daemon mode.)

File: runtime/doc/nvim_clipboard.txt [hunk 3]
```diff
@@ -29,4 +35,77 @@ following option:
 See 'clipboard' for details and more options.
 
 ==============================================================================
+2. X11 selection mechanism		      *nvim-clipboard-x11* *x11-selection*
+
+X11 provides two basic types of global store, selections and cut-buffers,
```
Comment (bfredl) [inline runtime/doc/nvim_clipboard.txt]: I don't think this should mention cut-buffers so prominently, and give reader (unfamiliar with the inner workings of X11) the impression that they are on equal footing with selections. They are generally considered obsolete (as they don't support formats, for instance) and many modern applications simply don't use them at all.

**Comment by fmoralesc** [comment]
@Pyrohh seconding @lucc's comments, I think it's best to move the `guipty` syntax to the vimOnlyOption group, like I did in #2902 (maybe rebase this over that?), or even leave that for now and I can add it there.

**Comment by Pyrohh** [comment]
Thanks for taking a look everyone.

@fmoralesc @lucc

> seconding @lucc's comments, I think it's best to move the guipty syntax to the vimOnlyOption group, like I did in #2902 (maybe rebase this over that?), or even leave that for now and I can add it there.

Sure, I dropped the `syntax/vim.vim` changes, so I shouldn't need to rebase on top of your PR. The things I removed (so you can add them to your PR) were `(no|inv|)guipty` and `VisualNOS`.

**Comment by Pyrohh** [comment]
Just rebased and fixed the merge conflict. Forgot to do it earlier, sorry.

**Comment by fmoralesc** [comment]
OK, I added those changes to my PR ;)

File: runtime/doc/nvim_clipboard.txt (hunk 2 above)
Comment (Pyrohh) [inline runtime/doc/nvim_clipboard.txt]: > Instead, to keep any owned selection alive after nvim exists, we should detach the clipboard owning xsel job ( or alternatively relaunch xsel in daemon mode.)

Maybe, but would that introduce the possibility of multiple `xsel` daemons being active simultaneously? Anyways, I'm just updating the docs, so I'll remove this section.

**Comment by Pyrohh** [comment]
@bfredl Thanks for taking a look, I reduced the size of the X11 clipboard section a lot. I rewrote some things so people should still have enough info to use the X11 clipboard providers, and (roughly) how they work.

If it's too sparse, please let me know.

File: runtime/doc/nvim_clipboard.txt (hunk 2 above)
Comment (bfredl) [inline runtime/doc/nvim_clipboard.txt]: : +1: we can try implement retaining clipboard when/if someone actually asks about it.

> but would that introduce the possibility of multiple xsel daemons being active simultaneously? 

No, `xsel` exits immediately when it loses ownership, no matter the process management.

**Comment by bfredl** [comment]
@Pyrohh a few suggestions, otherwise LGTM :+1:

**Comment by Pyrohh** [comment]
Okey, repushed once more with suggestions by @bfredl and @fmoralesc. I'll merge this in a day or two if no more comments.

**Comment by lucc** [comment]
You are deleting the tags for xdefaults, athena and motif in the gui_x11.txt file. If I grep for these in the repository I get a lot of matches (not for the tags but the words). These references should maybe be cleared up in a follow up of this PR.

**Comment by Pyrohh** [comment]
> These references should maybe be cleared up in a follow up of this PR.

Of course, I'll add it to my list.

**Pull request merged**